### ISSUE TITLE
How is the 'chembl_split.json' file generated?

### ISSUE THREAD

**Issue by model-fps** [BFeng14/ActFound#15]
Hi,author:
I noticed that the 'chembl_preprocess.py' file contains the code to preprocess the CHEMBL dataset and generate the 'chembl_processed_chembl32.csv' file, but I did not see the code to generate the 'chembl_split.json' file in the 'data_chembl_Assay_reg.py' file. How is the dataset split and saved as the 'chembl_split.json' file?
Looking forward to your reply!

**Comment by BFeng14** [comment]
Hi Leslie, the chembl_split.json contains the train, valid, test split of ChEMBL dataset, and they are simply generated randomly~ I wrote the code in my ipynb notebook and I cannot find the it now, and it just contains a random shuffle followed with split 500 for valid and 500 for test.

**Comment by model-fps** [comment]
Ok,I see.Thank for your reply!
Now I use a dataset I created myself, which contains 965 train assays and 437 test assays. There is no validation set. The maximum number of ligands in the assays is 282, and the average number of ligands is 75.
 However, when I modified and ran the `main_reg.py `code, I found that the output `meta_batch_loss['loss']` result was `nan`.
```
number of training set : 965
number of testing set : 437
100%|███████████████████████████████████████| 1402/1402 [14:17<00:00,  1.63it/s]
train_cnt:965, test_cnt:437
train_cnt:965, test_cnt:437
282 75.7867332382311
10
/home/Yq/anaconda3/envs/ActFound_e3fp_env/lib/python3.8/site-packages/numpy/lib/function_base.py:2854: RuntimeWarning: invalid value encountered in divide
  c /= stddev[:, None]
/home/Yq/anaconda3/envs/ActFound_e3fp_env/lib/python3.8/site-packages/numpy/lib/function_base.py:2855: RuntimeWarning: invalid value encountered in divide
  c /= stddev[None, :]
epoch is: 0, mean rmse is: 6.234
epoch is: 0, r2: mean is: 0.000, median is: 0.000, cnt>0.3 is: 0.000
epoch is: 0, R2os: mean is: nan, median is: nan, cnt>0.3 is: 0.000
/home/Yq/anaconda3/envs/ActFound_e3fp_env/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:131: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
/home/Yq/anaconda3/envs/ActFound_e3fp_env/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:156: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
/home/Yq/anaconda3/envs/ActFound_e3fp_env/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:728: UserWarning: To get the last learning rate computed by the scheduler, please use `get_last_lr()`.
  warnings.warn("To get the last learning rate computed by the scheduler, "
Current meta_batch_loss: nan
Current meta_batch_loss: nan
Current meta_batch_loss: nan
Current meta_batch_loss: nan
Current meta_batch_loss: nan
```
 What is the reason for this? I have checked that the shape of the data output is fine. Which settings should I modify or which step of my steps is wrong?

**Comment by BFeng14** [comment]
Hi, I noticed that "r2: mean is: 0.000" and "R2os: mean is: nan" in your first validation, which might be problem in your training data (maybe all activities in an assay are the same?), you can try to print your datas and check them in the forward function of learning_system/system_actfound.py

**Comment by model-fps** [comment]
But when I train on the chemble dataset, the results are as follows：
```
35644
number of training set before filter: 33296
number of training set after filter: 32867
100%|█████████████████████████████████████| 33867/33867 [06:32<00:00, 86.27it/s]
38105 500 500
8103 213.0757703618463
cir_num:10
/home/Yq/anaconda3/envs/ActFound/lib/python3.8/site-packages/numpy/lib/function_base.py:2854: RuntimeWarning: invalid value encountered in divide
  c /= stddev[:, None]
/home/Yq/anaconda3/envs/ActFound/lib/python3.8/site-packages/numpy/lib/function_base.py:2855: RuntimeWarning: invalid value encountered in divide
  c /= stddev[None, :]
epoch is: 0, mean rmse is: 2.499
epoch is: 0, r2: mean is: 0.000, median is: 0.000, cnt>0.3 is: 0.000
epoch is: 0, R2os: mean is: nan, median is: nan, cnt>0.3 is: 0.000
/home/Yq/anaconda3/envs/ActFound/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:131: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
/home/Yq/anaconda3/envs/ActFound/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:156: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
/mnt/sda/yq/ActFound/learning_system/system_base.py:126: UserWarning: Anomaly Detection has been enabled. This mode will increase the runtime and should only be enabled for debugging.
  with torch.autograd.detect_anomaly():
/home/Yq/anaconda3/envs/ActFound/lib/python3.8/site-packages/torch/_tensor.py:1083: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at  /opt/conda/conda-bld/pytorch_1659484810403/work/build/aten/src/ATen/core/TensorBody.h:477.)
  return self._grad
/home/Yq/anaconda3/envs/ActFound/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:728: UserWarning: To get the last learning rate computed by the scheduler, please use `get_last_lr()`.
  warnings.warn("To get the last learning rate computed by the scheduler, "
epoch: 0, iter: 199, mse: 1.1263201236724854
epoch: 0, iter: 399, mse: 1.0083661079406738
```
`R2os: mean is: nan, median is: nan` Is this normal?

**Comment by BFeng14** [comment]
that seems wired and is not normal, please show me your running command, and your modifications to the code if possible

**Comment by model-fps** [comment]
I found that I used the wrong run command, so I retrained the model in the original ActFound project file using the following command：
```
FIXED_PARAM="--test_sup_num 16 --test_repeat_num 2 --begin_lrloss_epoch 50 --metatrain_iterations 80 --no_fep_lig "

CHEMBL_DIR="./checkpoints_all/checkpoints_chembl_nofep"
CHEMBL_RES="./test_results/result_indomain/chembl"
python -u main_reg.py --datasource=chembl --logdir ${CHEMBL_DIR}/checkpoint_chembl_actfound --model_name actfound --test_write_file ${CHEMBL_RES} ${FIXED_PARAM} > ./runlog_nofep/chembl_actfound.log &
```
But the running process is very slow, and I haven't seen the output information yet, so I can't judge whether the result is correct. Could you please tell me what configuration of GPU you are running the code on? I am using a RTX3090 GPU.

**Comment by BFeng14** [comment]
I used RTX3090 GPU too and the training process was very slow, which requires 3 days roughly. (training meta-learning is very slow, I tried to speed up using multi-gpu training but end up with decrease in performance ... ), you may change test_repeat_num to 1 (number of repeating sampiling in validation set) or manually change the "Print_Iter" in the training code, which is set to be 200.

**Comment by model-fps** [comment]
Thanks for your reply！
